static/src/javascripts/projects/commercial/modules/dfp/define-slot.ts

```diff
@@ -85,21 +113,30 @@ const allowSafeFrameToExpand = (slot) => {
 	return slot;
 };
 
-const defineSlot = (adSlotNode, sizes) => {
-	const slotTarget = adSlotNode.getAttribute('data-name');
-	const sizeOpts = getSizeOpts(sizes);
+const defineSlot = (
+	adSlotNode: HTMLElement,
+	sizeMapping: SizeMapping,
+): { slot: googletag.Slot; slotReady: Promise<void> } => {
+	const slotTarget = adSlotNode.getAttribute('data-name') as SlotName;
+	const sizeOpts = getSizeOpts(sizeMapping);
 	const id = adSlotNode.id;
-	let slot;
+	let slot: googletag.Slot | null;
```

Comment:
This might be super pedantic but is this also possibly `undefined` at the point we first declare `slot`?

```suggestion
	let slot: googletag.Slot | null | undefined;
```

Reply:
Got it, that makes sense 👍